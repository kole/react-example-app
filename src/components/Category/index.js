import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import DOMPurify from 'dompurify';

const Category = styled.div`
    font-size: 1.4em;
    font-weight: bold;
`;

const CategoryComponent = (props) => {
    const { categoryText } = props;

    // Lets clean the encoded HTML before we unsafely pass it to the DOM
    const cleanHTML = DOMPurify.sanitize(categoryText);

    return <Category dangerouslySetInnerHTML={{__html: cleanHTML}}></Category>;
}

CategoryComponent.propTypes = {
    categoryText: PropTypes.string.isRequired
}

export default CategoryComponent;
