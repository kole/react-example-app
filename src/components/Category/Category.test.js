/* eslint-env jest */
import React from 'react';
import renderer from 'react-test-renderer';
import Category from './index.js';

describe('Category', () => {
    test('Snapshot', () => {
        const component = renderer
            .create(<Category categoryText="Test category" />)
            .toJSON();
        expect(component).toMatchSnapshot();
    });
});
