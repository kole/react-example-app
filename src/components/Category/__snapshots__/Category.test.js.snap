// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Category Snapshot 1`] = `
<div
  className="sc-bdVaJa gdwIIg"
  dangerouslySetInnerHTML={
    Object {
      "__html": "Test category",
    }
  }
/>
`;
